client/rest-high-level/src/main/java/org/elasticsearch/client/IndexLifecycleClient.java

```diff
@@ -39,6 +40,35 @@
         this.restHighLevelClient = restHighLevelClient;
     }
 
+    /**
+     * Delete a lifecycle definition
+     * See <a href="https://fix-me-when-we-have-docs.com">
+     * the docs</a> for more.
+     * @param request the request
+     * @param options the request options (e.g. headers), use {@link RequestOptions#DEFAULT} if nothing needs to be customized
+     * @return the response
+     * @throws IOException in case there is a problem sending the request or parsing back the response
+     */
+    public AcknowledgedResponse deleteLifecycle(DeleteLifecyclePolicyRequest request,
+                                                RequestOptions options) throws IOException {
+        return restHighLevelClient.performRequestAndParseEntity(request, RequestConverters::deleteLifecycle, options,
+            AcknowledgedResponse::fromXContent, emptySet());
+    }
+
+    /**
+     * Asynchronously delete a lifecycle definition
+     * See <a href="https://fix-me-when-we-have-docs.com">
+     * the docs</a> for more.
+     * @param request the request
+     * @param options the request options (e.g. headers), use {@link RequestOptions#DEFAULT} if nothing needs to be customized
+     * @param listener the listener to be notified upon request completion
+     */
+    public void deleteLifecycleAsync(DeleteLifecyclePolicyRequest request, RequestOptions options,
```

Comment:
In the same spirit as @dakrone's comment about renaming the request object to `DeleteLifecyclePolicyRequest` I think we should rename these methods to `deleteLifecyclePolicy` and `deleteLifecyclePolicyAsync`?